chain/network/src/network_protocol/mod.rs

```diff
@@ -52,7 +53,14 @@ pub struct AccountData {
 pub const MAX_ACCOUNT_DATA_SIZE_BYTES: usize = 10000; // 10kB
 
 impl AccountData {
-    pub fn sign(self, signer: &dyn near_crypto::Signer) -> anyhow::Result<SignedAccountData> {
+    pub fn sign(self, signer: &dyn ValidatorSigner) -> anyhow::Result<SignedAccountData> {
+        if &self.account_id != signer.validator_id() {
+            anyhow::bail!(
+                "AccountData.account_id doesn't match the signer's account_id: got {}, want {}",
+                self.account_id,
+                signer.validator_id()
+            );
+        }
```

Comment:
Should this be an assert? This feels like a clear deterministic programming error.

Reply:
It is likely to be a programming error, but it should be the caller who should do the panic (via unwrap).

Reply:
nvm, that would make sense if there were no valid runtime errors. I'll make it an assert.

Reply:
>It is likely to be a programming error, but it should be the caller who should do the panic (via unwrap).

Yeah, if this was a single error condition, and if we didn't use `anyhow`, that indeed would be better.

Reply:
done